core/opendaq/device/include/opendaq/device_impl.h

```diff
@@ -1581,6 +1619,17 @@ void GenericDevice<TInterface, Interfaces...>::deserializeCustomObjectValues(con
         this->template deserializeDefaultFolder<ISyncComponent>(serializedObject, context, factoryCallback, syncComponent, "Synchronization");
     }
 
+
+    userLock.reset();
+
+    if (serializedObject.hasKey("Locked"))
+    {
+        const bool isLocked = serializedObject.readBool("Locked");
+
+        if (isLocked)
+            userLock = nullptr;
```

Comment:
Right now if we serialize and deserialize device, we preserve lock status but loose the user who locked the device.

On the client side this is not a problem because we only care about current lock status of the device. The user validation is done on the server side when Lock / Unlock RPC is triggered.

It should be possible to preserve the user if we serialize the username and retrieve the user instance from authentication provider in the deserialize method. For this, a new method IAuthenticationProvider::getUser(IString username) should be added. IUserLock can also be also added for nicer serialization.